lib/AppInfo/Application.php

```diff
@@ -25,84 +25,56 @@
 namespace OCA\Text\AppInfo;
 
 use OCA\Text\DirectEditing\TextDirectEditor;
+use OCA\Text\Listeners\FilesSharingLoadAdditionalScriptsListener;
+use OCA\Text\Listeners\LoadViewerListener;
+use OCA\Text\Listeners\RegisterDirectEditorEventListener;
 use OCA\Viewer\Event\LoadViewer;
 use OCP\AppFramework\App;
+use OCP\AppFramework\Bootstrap\IBootContext;
+use OCP\AppFramework\Bootstrap\IBootstrap;
+use OCP\AppFramework\Bootstrap\IRegistrationContext;
 use OCP\DirectEditing\RegisterDirectEditorEvent;
 use OCP\EventDispatcher\IEventDispatcher;
 use OCP\IInitialStateService;
 
-class Application extends App {
-
-
+class Application extends App implements IBootstrap {
 	const APP_NAME = 'text';
 
-	/** @var IInitialStateService */
-	private $initialStateService;
-	/**
-	 * @var \OCP\IUserSession
-	 */
-	private $userSession;
-	/**
-	 * @var \OCP\IConfig
-	 */
-	private $config;
-
-
-	/**
-	 * Application constructor.
-	 *
-	 * @param array $params
-	 * @throws \OCP\AppFramework\QueryException
-	 */
 	public function __construct(array $params = []) {
 		parent::__construct(self::APP_NAME, $params);
+	}
 
-		$container = $this->getContainer();
-		$server = $container->getServer();
+	public function register(IRegistrationContext $context): void {
+		$context->registerEventListener(RegisterDirectEditorEvent::class, RegisterDirectEditorEventListener::class);
+		$context->registerEventListener(LoadViewer::class, LoadViewerListener::class);
+		$context->registerEventListener('OCA\Files_Sharing::loadAdditionalScripts', FilesSharingLoadAdditionalScriptsListener::class);
+	}
+
+	public function boot(IBootContext $context): void {
+		$server = $context->getServerContainer();
 		/** @var IEventDispatcher $eventDispatcher */
 		$eventDispatcher = $server->query(IEventDispatcher::class);
-		$this->initialStateService = $server->query(IInitialStateService::class);
-		$this->userSession = $server->getUserSession();
-		$this->config = $this->getContainer()->getServer()->getConfig();
-
-		$eventDispatcher->addListener(RegisterDirectEditorEvent::class, function (RegisterDirectEditorEvent $event) use ($container) {
-			$editor = $container->query(TextDirectEditor::class);
-			$event->register($editor);
-		});
-
-		$eventDispatcher->addListener(LoadViewer::class, function () {
-			\OCP\Util::addScript('text', 'viewer');
-			\OCP\Util::addStyle('text', 'icons');
-		});
+		$initialStateService = $server->query(IInitialStateService::class);
+		$userSession = $server->getUserSession();
+		$config = $server->getConfig();
 
-		if ($this->userSession->isLoggedIn()) {
-			$eventDispatcher->addListener('OCA\Files::loadAdditionalScripts', function () {
+		if ($userSession->isLoggedIn()) {
+			$eventDispatcher->addListener('OCA\Files::loadAdditionalScripts', function () use ($config, $userSession, $initialStateService) {
```

Comment:
For this one it would be super awesome to have a dedicated "if user is logged in" event to listen for, then we could move this as well.

Reply:
cc @rullzer to avoid the user session creation.

Reply:
If only somebody had the same idea... https://github.com/nextcloud/server/blob/master/apps/files/lib/Event/LoadAdditionalScriptsEvent.php

Reply:
Just use that
and reverse the logic?

Reply:
Always listen. and when it is emmitted check if the user is logged in

Reply:
I just noticed that this wasn't really used how it should be. Let me refine it a little bit.